pkg/controllers/termination/controller.go

```diff
@@ -77,24 +77,37 @@ func (c *Controller) Finalize(ctx context.Context, node *v1.Node) (reconcile.Res
 		return reconcile.Result{}, fmt.Errorf("cordoning node, %w", err)
 	}
 	if err := c.terminator.Drain(ctx, node); err != nil {
-		if terminator.IsNodeDrainError(err) {
-			c.recorder.Publish(terminatorevents.NodeFailedToDrain(node, err))
-			return reconcile.Result{Requeue: true}, nil
+		if !terminator.IsNodeDrainError(err) {
+			return reconcile.Result{}, fmt.Errorf("draining node, %w", err)
 		}
-		return reconcile.Result{}, fmt.Errorf("draining node, %w", err)
+		c.recorder.Publish(terminatorevents.NodeFailedToDrain(node, err))
+		// If the underlying machine no longer exists.
```

Comment:
The code as-is doesn't prevent you from hammering the API since if you get a node drain error over and over again you'll keep re-checking this condition.

Reply:
True. let me know what you think now